
From: devmfc <devmfc@mvand.net>
Subject: [PATCH 16/32] [DEVMFC] soc: amlogic: meson-gx-socinfo: Modify mask to
 add an additional S905X3 revision

Some android box with a S905X3 SoC had a pack_id of 1. Modify pack_mask
to detect that revision as a S905X3 as well.
---
 drivers/soc/amlogic/meson-gx-socinfo.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/amlogic/meson-gx-socinfo.c b/drivers/soc/amlogic/meson-gx-socinfo.c
index 7549f1644e..7944b8a186 100644
--- a/drivers/soc/amlogic/meson-gx-socinfo.c
+++ b/drivers/soc/amlogic/meson-gx-socinfo.c
@@ -77,8 +77,8 @@ static const struct meson_gx_package_id {
 	{ "A311D", 0x29, 0x10, 0xf0 },
 	{ "S922X", 0x29, 0x40, 0xf0 },
 	{ "S905D3", 0x2b, 0x4, 0xf5 },
-	{ "S905X3", 0x2b, 0x5, 0xf5 },
-	{ "S905X3", 0x2b, 0x10, 0x3f },
+	{ "S905X3", 0x2b, 0x1, 0xf1 },
+	{ "S905X3", 0x2b, 0x10, 0x37 },
 	{ "S905D3", 0x2b, 0x30, 0x3f },
 	{ "A113L", 0x2c, 0x0, 0xf8 },
 	{ "S805X2", 0x37, 0x2, 0xf },
-- 
2.46.0

From: devmfc <devmfc@mvand.net>
Subject: [PATCH 17/32] [DEVMFC] soc: amlogic: meson-gx-socinfo: Add some
 missing SoC ID's and names

Add some missing SoC ID's and their names. Based on recent vendor kernel.
---
 drivers/soc/amlogic/meson-gx-socinfo.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/soc/amlogic/meson-gx-socinfo.c b/drivers/soc/amlogic/meson-gx-socinfo.c
index 7944b8a186..a44d2f2b9d 100644
--- a/drivers/soc/amlogic/meson-gx-socinfo.c
+++ b/drivers/soc/amlogic/meson-gx-socinfo.c
@@ -41,6 +41,11 @@ static const struct meson_gx_soc_id {
 	{ "G12B", 0x29 },
 	{ "SM1", 0x2b },
 	{ "A1", 0x2c },
+	{ "TL1", 0x2e },
+	{ "C1", 0x30 },
+	{ "SC2", 0x32 },
+	{ "C2", 0x33 },
+	{ "T5", 0x34 },
 	{ "T7", 0x36 },
 	{ "S4", 0x37 },
 	{ "A5", 0x3c },
@@ -81,7 +86,9 @@ static const struct meson_gx_package_id {
 	{ "S905X3", 0x2b, 0x10, 0x37 },
 	{ "S905D3", 0x2b, 0x30, 0x3f },
 	{ "A113L", 0x2c, 0x0, 0xf8 },
+	{ "S905X4", 0x32, 0x2, 0xff },
 	{ "S805X2", 0x37, 0x2, 0xf },
+	{ "S905W2", 0x37, 0x6, 0xff },
 	{ "C308L", 0x3d, 0x1, 0xf },
 	{ "A311D2", 0x36, 0x1, 0xf },
 	{ "A113X2", 0x3c, 0x1, 0xf },
-- 
2.46.0

From: devmfc <devmfc@mvand.net>
Subject: [PATCH 18/32] [DEVMFC] soc: amlogic: meson-gx-socinfo: Add S905L-3
 SoC ID and name

---
 drivers/soc/amlogic/meson-gx-socinfo.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/soc/amlogic/meson-gx-socinfo.c b/drivers/soc/amlogic/meson-gx-socinfo.c
index a44d2f2b9d..70fe1ac8c4 100644
--- a/drivers/soc/amlogic/meson-gx-socinfo.c
+++ b/drivers/soc/amlogic/meson-gx-socinfo.c
@@ -39,6 +39,7 @@ static const struct meson_gx_soc_id {
 	{ "TXHD", 0x27 },
 	{ "G12A", 0x28 },
 	{ "G12B", 0x29 },
+	{ "GXLX2", 0x2a },
 	{ "SM1", 0x2b },
 	{ "A1", 0x2c },
 	{ "TL1", 0x2e },
@@ -81,6 +82,7 @@ static const struct meson_gx_package_id {
 	{ "S905X2", 0x28, 0x40, 0xf0 },
 	{ "A311D", 0x29, 0x10, 0xf0 },
 	{ "S922X", 0x29, 0x40, 0xf0 },
+	{ "S905L-3", 0x2a, 0xc5, 0xff },
 	{ "S905D3", 0x2b, 0x4, 0xf5 },
 	{ "S905X3", 0x2b, 0x1, 0xf1 },
 	{ "S905X3", 0x2b, 0x10, 0x37 },
-- 
2.46.0


From: devmfc <devmfc@mvand.net>
Subject: [PATCH 30/32] [DEVMFC] soc: amlogic: meson-gx-socinfo: Add S905Y4 SoC
 ID and name

---
 drivers/soc/amlogic/meson-gx-socinfo.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/soc/amlogic/meson-gx-socinfo.c b/drivers/soc/amlogic/meson-gx-socinfo.c
index 70fe1ac8c4..ec5b7a9715 100644
--- a/drivers/soc/amlogic/meson-gx-socinfo.c
+++ b/drivers/soc/amlogic/meson-gx-socinfo.c
@@ -90,6 +90,7 @@ static const struct meson_gx_package_id {
 	{ "A113L", 0x2c, 0x0, 0xf8 },
 	{ "S905X4", 0x32, 0x2, 0xff },
 	{ "S805X2", 0x37, 0x2, 0xf },
+	{ "S905Y4", 0x37, 0x3, 0xff },
 	{ "S905W2", 0x37, 0x6, 0xff },
 	{ "C308L", 0x3d, 0x1, 0xf },
 	{ "A311D2", 0x36, 0x1, 0xf },
-- 
2.46.0

